build_defs/arch_tests.bzl

```diff
@@ -0,0 +1,56 @@
+"""Generated unittests to verify that a binary is built for the expected architecture."""
+
+load("//build_defs:internal_shell.bzl", "inline_sh_test")
+
+def _arch_test_impl(
+        name,
+        platform,
+        file_platform,
+        bazel_binaries = [],
+        system_binaries = [],
+        **kwargs):
+    """
+    Bazel rule to verify that a Bazel or system binary is built for the
+    aarch64 architecture.
```

Comment:
My intention was that we would create one of these for each architecture we support.  I just added aarch64 and x64_64 for now because we only have test coverage there (another follow up task I'm writing up).  I'm not sure if `file` provides the OS, but if it does we could definitely use this for mac/windows too